### DISCUSSION TITLE
Text filter delay in onDelayedChange event

### DISCUSSION THREAD

**Discussion by pmosconi** [cloudscape-design/components#789] [Q&A] [answered]
Hi,
is it possible to configure the delay that triggers onDelayedChanged event in TextFilter? Or maybe is there a way to apply a debounce function?
I'm using it for server side table filtering and it looks very short, around 100ms probably which causes multiple calls to api that could be reduced.
Thanks

**Comment by taheramr** [comment, marked as answer]
Hello,

You can apply a debounce function that debounces the requests to your server API instead of depending on the debounce of a text input, you could have a look at this guide for starters https://plainenglish.io/blog/implementing-debouncing-in-react
